app/src/main/java/org/openobservatory/ooniprobe/common/TaskExecutor.kt

```diff
@@ -0,0 +1,56 @@
+package org.openobservatory.ooniprobe.common
+
+import android.os.Handler
+import android.os.Looper
+import java.util.concurrent.Callable
+import java.util.concurrent.Executors
+import java.util.concurrent.Future
+
+abstract class ProgressTask<P, R> {
+    abstract fun runTask(progressToken: OnTaskProgressUpdate<P>): R
+}
+
+typealias Task<R> = Callable<R>
+
+typealias OnTaskProgressUpdate<P> = (P) -> Unit
+
+typealias OnTaskComplete<R> = (R) -> Unit
+
+class TaskExecutor {
```

Comment:
Please, document what this class is about. (I also have some reservations into having a package called `common` where we're putting all sort of files; obviously, this is not something we can change right now, but I look forward to a moment in time where we're trying to avoid `common` and we have a more defined architecture.)

Reply:
Yeah. There is a need for that. I don't think this PR is the best place to start such changes

Reply:
Sure, not right now, but I wanted to remind ourselves there's a need for that!